tree/ntuple/v7/src/RDaos.cxx

```diff
@@ -107,47 +116,52 @@ ROOT::Experimental::Detail::RDaosObject::~RDaosObject()
 int ROOT::Experimental::Detail::RDaosObject::Fetch(FetchUpdateArgs &args)
 {
    args.fIods[0].iod_size = (daos_size_t)DAOS_REC_ANY;
-   return daos_obj_fetch(fObjectHandle, DAOS_TX_NONE,
-                         DAOS_COND_DKEY_FETCH | DAOS_COND_AKEY_FETCH,
-                         &args.fDistributionKey, 1, args.fIods, args.fSgls, nullptr, args.fEv);
+   return daos_obj_fetch(fObjectHandle, DAOS_TX_NONE, DAOS_COND_DKEY_FETCH | DAOS_COND_AKEY_FETCH,
+                         &args.fDistributionKey, 1, args.fIods, args.fSgls, nullptr, args.GetEventPointer());
 }
 
 int ROOT::Experimental::Detail::RDaosObject::Update(FetchUpdateArgs &args)
 {
-   return daos_obj_update(fObjectHandle, DAOS_TX_NONE, 0, &args.fDistributionKey, 1,
-                          args.fIods, args.fSgls, args.fEv);
+   return daos_obj_update(fObjectHandle, DAOS_TX_NONE, 0, &args.fDistributionKey, 1, args.fIods, args.fSgls,
+                          args.GetEventPointer());
 }
 
-
 ////////////////////////////////////////////////////////////////////////////////
 
-
-ROOT::Experimental::Detail::RDaosContainer::DaosEventQueue::DaosEventQueue(std::size_t size)
-  : fSize(size), fEvs(std::unique_ptr<daos_event_t[]>(new daos_event_t[size]))
+ROOT::Experimental::Detail::RDaosEventQueue::RDaosEventQueue()
 {
-   daos_eq_create(&fQueue);
-   for (std::size_t i = 0; i < fSize; ++i)
-      daos_event_init(&fEvs[i], fQueue, nullptr);
+   if (int ret = daos_eq_create(&fQueue) < 0)
+      throw RException(R__FAIL("daos_eq_create: error: " + std::string(d_errstr(ret))));
 }
 
-ROOT::Experimental::Detail::RDaosContainer::DaosEventQueue::~DaosEventQueue() {
-   for (std::size_t i = 0; i < fSize; ++i)
-      daos_event_fini(&fEvs[i]);
+ROOT::Experimental::Detail::RDaosEventQueue::~RDaosEventQueue()
+{
    daos_eq_destroy(fQueue, 0);
 }
 
-int ROOT::Experimental::Detail::RDaosContainer::DaosEventQueue::Poll() {
-   auto evp = std::unique_ptr<daos_event_t*[]>(new daos_event_t*[fSize]);
-   std::size_t n = fSize;
-   while (n) {
-      int c;
-      if ((c = daos_eq_poll(fQueue, 0, DAOS_EQ_WAIT, n, evp.get())) < 0)
-         break;
-      n -= c;
-   }
-   return n;
+int ROOT::Experimental::Detail::RDaosEventQueue::InitializeEvent(daos_event_t *ev_ptr, daos_event_t *parent_ptr)
+{
+   return daos_event_init(ev_ptr, fQueue, parent_ptr);
+}
+
+int ROOT::Experimental::Detail::RDaosEventQueue::FinalizeEvent(daos_event_t *ev_ptr)
+{
+   return daos_event_fini(ev_ptr);
 }
 
+int ROOT::Experimental::Detail::RDaosEventQueue::WaitOnParentBarrier(daos_event_t *ev_ptr)
+{
+   int err;
+   bool flag = false;
+
+   if ((err = daos_event_parent_barrier(ev_ptr) < 0))
+      return err;
+
+   // Waiting call on event completion
```

Comment:
Not sure if the comment adds something given that `DAOS_EQ_WAIT` below, which suggests blocking behavior.